### PR TITLE
Support Set "app_no_delay" Attributes

### PR DESCRIPTION
**Pull request by jococi**

### CODE REVIEW

File: session.go [hunk 1]
```diff
@@ -530,11 +554,72 @@ LOOP:
...
+
+									}
+								default:
+									return
+								}
+							}
+						}
+						f()
+						if s.delay > 0 {
+							if _, ok := s.Connection.(*gettyUDPConn); ok {
```
Comment (wongoo) [inline session.go]: duplicated code for sending packages can be packeted to a new method

File: session.go [hunk 2]
```diff
@@ -530,11 +554,72 @@ LOOP:
...
+										s.delay = 0
+										s.dBuff = []byte{}
+
+									}
+								default:
+									return
+								}
+							}
+						}
+						f()
```
Comment (wongoo) [inline session.go]: the f() is called once, not need to package as a method

File: session.go [hunk 3]
```diff
@@ -530,11 +554,72 @@ LOOP:
...
 		case outPkg = <-s.wQ:
 			if flag {
-				if err = s.writer.Write(s, outPkg); err != nil {
+				var b []byte
+				if b, err = s.writer.Write(s, outPkg); err != nil {
 					log.Error("%s, [session.handleLoop] = error{%s}", s.sessionToken(), jerrors.ErrorStack(err))
 					s.stop()
 					flag = false
 					// break LOOP
+				} else {
```
Comment (wongoo) [inline session.go]: lots of if/else

**Comment by wongoo** [comment]
this pr is about batch write, seems conflict with https://github.com/AlexStocks/getty/pull/23

**Comment by AlexStocks** [comment]
I think this pr changed so much codes, Especially the handleLoop which is so ugly.

**Pull request closed without merging**